Command/AnonymizeDataCommand.php

```diff
@@ -0,0 +1,113 @@
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Class AnonymizeDataCommand
+ *
+ * @author Benoit Mazière <benoit.maziere@ekino.com>
+ */
+final class AnonymizeDataCommand extends Command
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected static $defaultName = 'ekino-data-protection:anonymize';
+
+    /**
+     * @var AnonymizedMetadataBuilder
+     */
+    protected $anonymizedMetadataBuilder;
+
+    /**
+     * @var AnonymizedMetadataValidator
+     */
+    protected $anonymizedMetadataValidator;
+
+    /**
+     * @var AnonymizedQueryBuilder
+     */
+    protected $anonymizedQueryBuilder;
+
+    public function __construct(
+        AnonymizedMetadataBuilder $anonymizedMetadataBuilder,
+        AnonymizedMetadataValidator $anonymizedMetadataValidator,
+        AnonymizedQueryBuilder $anonymizedQueryBuilder
+    )
+    {
+        parent::__construct();
+
+        $this->anonymizedMetadataBuilder   = $anonymizedMetadataBuilder;
+        $this->anonymizedMetadataValidator = $anonymizedMetadataValidator;
+        $this->anonymizedQueryBuilder      = $anonymizedQueryBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setDescription('Anonymize database based on entities annotations')
+            ->addOption('force', null, InputOption::VALUE_NONE, 'Set this parameter to execute this action')
+            ->setHelp('Usage: `bin/console ekino-data-protection:anonymize`')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $output->writeln(sprintf('<info>Anonymization starts</info>'));
+
+        $anonymizedMetadatas = $this->anonymizedMetadataBuilder->build();
+        $queries             = [];
+
+        foreach ($anonymizedMetadatas as $anonymizedMetadata) {
+            $this->anonymizedMetadataValidator->validate($anonymizedMetadata);
+            $queries[] = $this->anonymizedQueryBuilder->buildQuery($anonymizedMetadata);
+        }
+
+        if (!$input->getOption('force')) {
+            $output->writeln('<error>ATTENTION:</error> This operation should not be executed in a production environment.');
+            $output->writeln('');
+            $output->writeln('<info>Would annoymize your database according to your configuration.</info>');
+            $output->writeln('Please run the operation with --force to execute');
+            $output->writeln('<error>Some data will be lost/anonymized!</error>');
+
+            // display queries
+            $output->writeln('<error>Following queries have been built and will be executed:</error>');
+
+            foreach ($queries as $query) {
+                $output->writeln(sprintf('<info>%s</info>', $query));
+            }
+
+            return 0;
+        }
+
+        // display & execute queries
```

Comment:
please ask confirmation before

Reply:
I don't see when the queries are actually executed? 🤔